client/src/components/AddHome.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { addHome } from "../redux/homesSlice";
+
+const AddHome = () => {
```

Comment:
- [ ] When I try to add a new house, this happens:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/ef86f640-782a-4397-9a86-345b8939e466)

This is because CORS is blocking the request.

## Let's solve the problem

### In config/application.rb:

Can you notice the syntax error?

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/673f4d04-84ae-4dc0-aa62-b5f2e2c1c3ac)

**Explanation:** _When you use i% before an array, it means "Non-interpolated Array of symbols, separated by whitespace".  You don't need the colons anymore.  Ruby now knows these elements are symbols.  If you add colons, you are effectively adding _double_ colons._

[Here](https://stackoverflow.com/questions/39803299/using-i-and-i-symbol-array-literal) is the complete explanation.

This seems to be the same issue as in the routes file.

Let's change the `methods:`keyword argument for `resource '*', headers: :any, methods: %i[get post put patch delete options]`

Like this:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/9f507107-a360-4c5e-afe8-2cdc03c87420)

Now, let's close the server and re-run it, so our CORS correction is taken into account, and try to create a new home:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/4ad4e82c-7bb6-4979-be3e-fefe36670335)

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/9cc36295-b3fd-413a-9631-eaa50bdd82d9)

It works!!!!

Kindly, fix this issue now that you have the context and a possible solution. 💪